sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -1109,21 +1138,37 @@ case class ShowCreateTableCommand(table: TableIdentifier)
           )
         }
 
-        if (tableMetadata.tableType == VIEW) {
-          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE")
-        }
-
         if ("true".equalsIgnoreCase(tableMetadata.properties.getOrElse("transactional", "false"))) {
           throw new AnalysisException(
             "SHOW CREATE TABLE doesn't support transactional Hive table. " +
               s"Please use `SHOW CREATE TABLE ${tableMetadata.identifier} AS SERDE` " +
               "to show Hive DDL instead.")
         }
 
-        convertTableMetadata(tableMetadata)
+        if (tableMetadata.tableType == VIEW) {
+          tableMetadata
+        } else {
+          convertTableMetadata(tableMetadata)
+        }
       }
 
-      val stmt = showCreateDataSourceTable(metadata)
+      val builder = StringBuilder.newBuilder
+
+      val stmt = if (tableMetadata.tableType == VIEW) {
```

Comment:
can we create a method to generate `CREATE VIEW`? So we can just call one method in the `if` branch.

Reply:
+1 for @cloud-fan 's advice. We had better have a function similar to `showCreateDataSourceTable`.

Reply:
Created a method like suggestion.